[PAL/Linux] Failed to declare '_r_debug@@PAL'  as default

## Description of the problem
The Graphene building failed due to an Error as follows
> make[3]: Entering directory '/home/gary/graphene/Pal/src/host/Linux'
>   [ db_eventfd.o ]
> /tmp/ccOTt36Q.s: Assembler messages:
> /tmp/ccOTt36Q.s: Error: invalid attempt to declare external version name as default in symbol `_r_debug@@PAL'
> Makefile:65: recipe for target 'db_eventfd.o' failed
> make[3]: *** [db_eventfd.o] Error 1

on platform

> VERSION="18.04.3 LTS (Bionic Beaver)"
> ID=ubuntu
> ID_LIKE=debian
> PRETTY_NAME="Ubuntu 18.04.3 LTS"
> gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
> GNU ld (GNU Binutils) 2.34.50.20200426

## Steps to reproduce
make

## Expected results
No any errors

## Actual results
reports

> make[3]: Entering directory '/home/gary/graphene/Pal/src/host/Linux'
>   [ db_eventfd.o ]
> /tmp/ccOTt36Q.s: Assembler messages:
> /tmp/ccOTt36Q.s: Error: invalid attempt to declare external version name as default in symbol `_r_debug@@PAL'
> Makefile:65: recipe for target 'db_eventfd.o' failed
> make[3]: *** [db_eventfd.o] Error 1
>

--- a/Pal/src/host/Linux/db_rtld.c
+++ b/Pal/src/host/Linux/db_rtld.c
@@ -28,26 +28,22 @@
 
 #include "elf-arch.h"
 
-/* This structure communicates dl state to the debugger.  The debugger
-   normally finds it via the DT_DEBUG entry in the dynamic section, but in
-   a statically-linked program there is no dynamic section for the debugger
-   to examine and it looks for this particular symbol name.  */
-struct r_debug g_pal_r_debug = { 1, NULL, (ElfW(Addr))&pal_dl_debug_state, RT_CONSISTENT, 0 };
-
-/* This function exists solely to have a breakpoint set on it by the
-   debugger.  The debugger is supposed to find this function's address by
-   examining the r_brk member of struct r_debug, but GDB 4.15 in fact looks
-   for this particular symbol name in the PT_INTERP file.  */
-
-/* The special symbol name is set as breakpoint in gdb */
-void __attribute__((noinline)) pal_dl_debug_state (void)
-{
+/* This function exists solely to have a breakpoint set on it by the debugger. The debugger is
+ * supposed to find this function's address by examining the r_brk member of struct r_debug, but GDB
+ * 4.15 in fact looks for this particular symbol name in the PT_INTERP file.  */
+static void __attribute__((noinline)) pal_dl_debug_state(void) {
     if (g_pal_sec._dl_debug_state)
         g_pal_sec._dl_debug_state();
 }
 
 extern __typeof(pal_dl_debug_state) _dl_debug_state
-    __attribute ((alias ("pal_dl_debug_state")));
+    __attribute((alias("pal_dl_debug_state")));
+
+/* This structure communicates dl state to the debugger.  The debugger normally finds it via the
+ * DT_DEBUG entry in the dynamic section, but in a statically-linked program there is no dynamic
+ * section for the debugger to examine and it looks for this particular symbol name.  */
+struct r_debug g_pal_r_debug = {1, NULL, (ElfW(Addr))&pal_dl_debug_state, RT_CONSISTENT, 0};
+symbol_version_default(g_pal_r_debug, _r_debug, PAL);
 
 void _DkDebugAddMap (struct link_map * map)
 {

--- a/Pal/src/host/Linux/pal_security.h
+++ b/Pal/src/host/Linux/pal_security.h
@@ -35,15 +35,6 @@ struct r_debug {
     ElfW(Addr) r_ldbase; /* Base address the linker is loaded at.  */
 };
 
-void pal_dl_debug_state(void);
-
-/* This structure communicates dl state to the debugger.  The debugger
-   normally finds it via the DT_DEBUG entry in the dynamic section, but in
-   a statically-linked program there is no dynamic section for the debugger
-   to examine and it looks for this particular symbol name.  */
-extern struct r_debug g_pal_r_debug;
-symbol_version_default(g_pal_r_debug, _r_debug, PAL);
-
 extern struct pal_sec {
     /* system variables */
     unsigned int process_id;

